packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -169,44 +181,64 @@ export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
 
   const checkAllocatedQuantities = () => {
     const values = getValues();
-    const allocatedQuantity = batchRows.reduce(
-      (total, batch) => (total += Number(values[batch.id] || 0)),
+    const allocatedUnits = batchRows.reduce(
+      (total, batch) =>
+        (total += Number(values[batch.id] || 0) * batch.packSize),
       Number(values['placeholder'] || 0)
     );
-
-    setIsAllocated(allocatedQuantity >= quantity && allocatedQuantity > 0);
+    setAllocated(Math.floor(allocatedUnits / packSize));
   };
 
   const allocateQuantities = () => {
     // if invalid quantity entered, don't allocate
-    if (quantity < 1 || Number.isNaN(quantity)) {
+    if (quantity < 1 || Number.isNaN(quantity) || !selectedItem) {
       return;
     }
     // if the selected item has no batch rows, allocate all to the placeholder
     if (batchRows.length === 0) {
       setValue('placeholder', quantity);
-      setIsAllocated(true);
+      setAllocated(quantity);
       return;
     }
 
-    let toAllocate = 0;
-    toAllocate += quantity;
+    // calculations are normalised to units
+    let toAllocate = quantity * packSize;
+    let batchAllocation = 0;
 
     batchRows.forEach(batch => {
-      const allocatedQuantity = Math.min(
-        toAllocate,
-        batch.availableNumberOfPacks * batch.packSize
-      );
-      toAllocate -= allocatedQuantity;
-      setValue(batch.id, allocatedQuantity);
+      batchAllocation = 0;
+      // skip bigger pack sizes
```

Comment:
I would have thought that if we were selecting pack sizes we would only assign to that specific pack size, rather than anything smaller? Maybe I don't understand the use case 🤷

Reply:
I think we probably should stick to the exact pack size and avoid the weirdness that I've introduced here. maybe also remove the pack size of 1 that is added to the list then, and only show available pack sizes? and then change placeholder input to be # packs not # units.. and then record the pack size when creating placeholder lines..